File: NearbyPlaces/Place.m
//
//  Place.m
//  NearbyPlaces
//
//  Created by Adrian Kubała on 14.02.2017.
//  Copyright © 2017 Adrian Kubała. All rights reserved.
//

#import "Place.h"

@implementation Place

-(instancetype)initWithName:(NSString *)name
                    address:(NSString *)address
                 coordinate:(CLLocationCoordinate2D)coordinate
                      photo:(UIImage *)photo
               userLocation:(CLLocationCoordinate2D)userLocation {
  self = [super init];
  
  if (self) {
    _name = name;
    _address = address;
    _coordinate = coordinate;
    _photo = photo;
    _distance = [self getDistanceFrom:coordinate to:userLocation];
  }
  
  return self;
}

-(NSInteger)getDistanceFrom:(CLLocationCoordinate2D)place to:(CLLocationCoordinate2D)otherPlace {
  CLLocation *firstLocation = [[CLLocation alloc] initWithLatitude:place.latitude longitude:place.longitude];
  CLLocation *secondLocation = [[CLLocation alloc] initWithLatitude:otherPlace.latitude longitude:otherPlace.longitude];
  
  int distanceInMeters = [secondLocation distanceFromLocation:firstLocation];
  return distanceInMeters;
}

@end

File: Places/Place.swift
//
//  Place.swift
//  Places
//
//  Created by Adrian on 15.09.2016.
//  Copyright © 2016 Adrian Kubała. All rights reserved.
//

import Foundation
import GooglePlaces

class Place {
  var name: String
  var address: String?
  var distance: Int = 0
  var coordinate: CLLocationCoordinate2D
  var photo: UIImage
  
  init(name: String, address: String?, coordinate: CLLocationCoordinate2D, photo: UIImage, userLocation: CLLocationCoordinate2D) {
    self.name = name
    self.address = address
    self.coordinate = coordinate
    self.photo = photo
    distance = getDistance(from: coordinate, to: userLocation)
  }
  
  func getDistance(from place: CLLocationCoordinate2D, to otherPlace: CLLocationCoordinate2D) -> Int {
    let firstLocation = CLLocation(latitude: place.latitude, longitude: place.longitude)
    let secondLocation = CLLocation(latitude: otherPlace.latitude, longitude: otherPlace.longitude)
    
    let distanceInMeters = secondLocation.distance(from: firstLocation)
    distance = Int(distanceInMeters)
    return distance
  }
}
